src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2349,10 +2351,10 @@ private void LexExcludedDirectivesAndTrivia(bool endIsActive, ref SyntaxListBuil
             CSharpSyntaxNode directive;
             var saveMode = _mode;
 
-            using (var dp = new DirectiveParser(this, _directives))
-            {
-                directive = dp.ParseDirective(isActive, endIsActive, afterFirstToken, afterNonWhitespaceOnLine);
-            }
+            _directiveParser ??= new DirectiveParser(this);
```

Comment:
>_directiveParser ??= new DirectiveParser(this); [](http://example.com/codeflow?start=12&length=47)

Might be nice if _xmlParser and _directiveParser construction/reinitialization had the exact same form.

Reply:
Kept them slightly different as reinitialize allocs on the doc comment case.  So it saved an allocation there.

Reply:
Resolved.